airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/analytics.py

```diff
@@ -0,0 +1,206 @@
+# to have the `dateRange`, `pivot`, and `pivotValue` be included as well.
+FIELDS_CHUNK_SIZE = 17
+# Number of days ahead for date slices, from start date.
+WINDOW_IN_DAYS = 30
+# List of adAnalyticsV2 fields available for fetch
+ANALYTICS_FIELDS_V2: List = [
+    "actionClicks",
+    "adUnitClicks",
+    "approximateUniqueImpressions",
+    "cardClicks",
+    "cardImpressions",
+    "clicks",
+    "commentLikes",
+    "comments",
+    "companyPageClicks",
+    "conversionValueInLocalCurrency",
+    "costInLocalCurrency",
+    "costInUsd",
+    "dateRange",
+    "externalWebsiteConversions",
+    "externalWebsitePostClickConversions",
+    "externalWebsitePostViewConversions",
+    "follows",
+    "fullScreenPlays",
+    "impressions",
+    "landingPageClicks",
+    "leadGenerationMailContactInfoShares",
+    "leadGenerationMailInterestedClicks",
+    "likes",
+    "oneClickLeadFormOpens",
+    "oneClickLeads",
+    "opens",
+    "otherEngagements",
+    "pivot",
+    "pivotValue",
+    "pivotValues",
+    "reactions",
+    "sends",
+    "shares",
+    "textUrlClicks",
+    "totalEngagements",
+    "videoCompletions",
+    "videoFirstQuartileCompletions",
+    "videoMidpointCompletions",
+    "videoStarts",
+    "videoThirdQuartileCompletions",
+    "videoViews",
+    "viralCardClicks",
+    "viralCardImpressions",
+    "viralClicks",
+    "viralCommentLikes",
+    "viralComments",
+    "viralCompanyPageClicks",
+    "viralExternalWebsiteConversions",
+    "viralExternalWebsitePostClickConversions",
+    "viralExternalWebsitePostViewConversions",
+    "viralFollows",
+    "viralFullScreenPlays",
+    "viralImpressions",
+    "viralLandingPageClicks",
+    "viralLikes",
+    "viralOneClickLeadFormOpens",
+    "viralOneClickLeads",
+    "viralOtherEngagements",
+    "viralReactions",
+    "viralShares",
+    "viralTotalEngagements",
+    "viralVideoCompletions",
+    "viralVideoFirstQuartileCompletions",
+    "viralVideoMidpointCompletions",
+    "viralVideoStarts",
+    "viralVideoThirdQuartileCompletions",
+    "viralVideoViews",
+]
+
+# Fields that are always present in fields_set chunks
+BASE_ANALLYTICS_FIELDS = ["dateRange", "pivot", "pivotValue"]
+
+
+def chunk_analytics_fields(
+    fields: List = ANALYTICS_FIELDS_V2,
+    base_fields: List = BASE_ANALLYTICS_FIELDS,
+    fields_chunk_size: int = FIELDS_CHUNK_SIZE,
+) -> Iterable[Mapping]:
+    """
+    Chunks the list of available fields into the chunks of equal size.
+    """
+    # Make chunks
+    chunks = list((fields[f : f + fields_chunk_size] for f in range(0, len(fields), fields_chunk_size)))
+    # Make sure base_fields are within the chunks
+    for chunk in chunks:
+        for field in base_fields:
+            if field not in chunk:
+                chunk.append(field)
+    return chunks
+
+
+def make_date_slices(start_date: str, end_date: str = None, window_in_days: int = WINDOW_IN_DAYS) -> Iterable[Mapping]:
```

Comment:
Please take into considaretion that currently you have  sich slices

(windows_in_days = 1 day)

2021.01.01 - 2021.01.02
2021.01.02 - 2021.01.03
2021.01.03 - 2021.01.04 

in case if API return data inclusive for data range  (returns data for start and for end data). then report for the same  days will be extracted in to diffrent slices - like results of  for 2021.01.02 dat will appear in first and in second slice.

Reply:
Of corse, this behaviour was tested before the implementation, it is working correctly as it is now. No duplications. But thank you for your considerations!